### DISCUSSION TITLE
Issues Importing Code

### DISCUSSION THREAD

**Discussion by Matt7283762** [community/community#27515] [Other Feature Feedback, Questions, & Ideas]
Hello, this is following up on an issue I posted about in the old forum (https://github.community/t/issues-importing-code/262900) on receiving an error when using the "Import code" button on github.com. It was recommended that I try manually cloning the repository.

After running "git clone --bare" on the existing private repository, I ran the following command to clone the repository into a new private repository, but receive an error:

```
git push --mirror https://github.com/Matt7283762/MotusAudereMeridian.git
Enumerating objects: 1133, done.
Counting objects: 100% (1133/1133), done.
Delta compression using up to 16 threads
Compressing objects: 100% (476/476), done.
error: RPC failed; curl 92 HTTP/2 stream 0 was not closed cleanly: CANCEL (err 8)
send-pack: unexpected disconnect while reading sideband packet
Writing objects: 100% (1133/1133), 7.29 GiB | 17.19 MiB/s, done.
Total 1133 (delta 650), reused 1133 (delta 650), pack-reused 0
fatal: the remote end hung up unexpectedly
Everything up-to-date
```

**Comment by airtower-luna** [comment]
See the [documentation about large files](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github):

> GitHub blocks pushes that exceed 100 MB.

I'm kinda surprised they let you get to 7.29 GiB before canceling the connection. :sweat_smile: 

If the push is large but comprised of small files, you should be able to push in chunks. If you have individual files bigger than the limits listed in the documentation you'll have to find some other option (LFS, or hosting elsewhere).

**Reply by Matt7283762** [reply]
How can I push it in chunks? What's the command?

**Reply by airtower-luna** [reply]
In most cases people use `git push` to push a local ref (branch or tag) to a remote one of the same name. But you can tell it to push something else, e.g. a random commit:

```
git push origin 1234...:my-branch
```

This would push the commit `1234...` to the remote branch `my-branch`. If you have a big commit history, you can use this to split it into chunks. Let's says you have a local commit history like this that's too big to push at once, that you want to push to the remote `main` branch (A is the initial commit, B somewhere in between, C the current HEAD):

```
A---o---...---B---...---o---C
```

Get the commit ID of B, and push it like in the example above:

```
git push origin <SHA-of-B>:main
```

Because you're only pushing part of the history the push is smaller. Pick an early enough commit for B, and (assuming you don't have any huge single commit/file) it can be small enough for the limit. Now the commits up to B are already on the remote, so you don't need to transfer them again in the next push. Repeat until you reach the current HEAD.

See also: https://git-scm.com/docs/git-push

**Reply by Matt7283762** [reply]
I still seem to be struggling with the syntax. I tried:
```
>git remote rename origin upstream
fatal: could not unset 'remote.upstream.fetch'
>git remote add origin https://github.com/Matt7283762/MotusAudereMeridian.git
>git push upstream acde3057ff9c8e899f4bbb6caffe04b0734cd484:refs/heads/origin
```
But that seems to have created a new branch on the old repository (now upstream)

**Reply by airtower-luna** [reply]
The push should simply be (substitute your target branch name for `your-branch`):

```
git push upstream acde3057ff9c8e899f4bbb6caffe04b0734cd484:your-branch
```

However I'd investigate the error while renaming, to make sure it doesn't cause issues later. I haven't seen that error message before, so my first step would be to check `git remote show` for both `origin` and `upstream`, see if the URLs are correct, and if branches are configured correctly. Also check the `fetch` setting for `upstream` in `.git/config`.
